src/hardware/hardware.cpp

```diff
@@ -371,12 +371,13 @@ void CAPTURE_AddImage(Bitu width, Bitu height, Bitu bpp, Bitu pitch, Bitu flags,
 		constexpr char value[] = "dosbox-staging " VERSION;
 		constexpr int num_text = 1;
 		static_assert(sizeof(keyword) < 80, "libpng limit");
-		png_text texts[num_text] = {};
-		texts[0].compression = PNG_TEXT_COMPRESSION_NONE;
-		texts[0].key = const_cast<png_charp>(keyword);
-		texts[0].text = const_cast<png_charp>(value);
-		texts[0].text_length = sizeof(value);
-		png_set_text(png_ptr, info_ptr, texts, num_text);
+		png_text texts[num_text] = {{
```

Comment:
(Assuming it can become `constexpr` too?)

Reply:
It can't because of libpng API design :(

libpng reuses this structure for reading and writing "text chunks" - char array fields use 
`’png_charp’ {aka ‘char*’}` type for key, text (and other fields, that I don't initialize here). These pointers do not pass ownership when setting (so I can get away with `const_cast` in here)… but they do prevent C++ from initializing this struct at compile time - testing consexpr results in error:

`hardware.cpp:379:4: error: ‘png_text [1]{png_text{-1, ((png_charp)(& keyword)), ((png_charp)(& value)), 19}}’ is not a constant expression`

:(

Reply:
And compilation failed in MSVC.

C++ "copied" (and renamed) this feature from C only with C++20 :(

https://en.cppreference.com/w/cpp/language/aggregate_initialization